### DISCUSSION TITLE
how to get msg type?

### DISCUSSION THREAD

**Discussion by mttbx** [protobuf-c/protobuf-c#604] [Q&A] [answered]
There's potential many message, how can I be sure which is the one I'm dealing with?

**Comment by ubx-luis** [comment, marked as answer]
Does oneof works for you?

message AllMyPossibleMsg{
  oneof message{
    messageType1 msg1 = 1;
    messageType2 msg2 = 2;
  }
}

This will add a variable in the AllMyPossibleMsg tied to an Enum indicating whether Type1 or Type2 of message was used...
Hope this helps!

**Comment by mttbx** [comment]
Yeah, that works, thank you!
